### PR TITLE
Fixes VSTS Bug 810343: Accessibility: Document View Tabs: Voice Over

### PR DESCRIPTION
**Pull request by mkrueger**
was not providing any information related to whether the "Source",
"Changes" etc are selected or not.

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/810343

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Tabstrip.cs [hunk 1]
```diff
@@ -345,36 +343,46 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
 		}
 
 		int focusedTab = -1;
+
+		int FocusedTab {
+			get => focusedTab;
+			set {
+				focusedTab = value;
+				if (focusedTab >= 0 && focusedTab < tabs.Count)
+					Accessible.Description = GettextCatalog.GetString ("Focused tab {0}", this.tabs[focusedTab].Label);
```
Comment (sevoku) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Tabstrip.cs]: I don't think that this is necessary, the native tab control (i.e. System Preferences > Keyboard) does not announce the currently selected tab when it receives focus. Also "Focused tab X" is misleading here, the focused tab is the one which has the VO focus, not the selected one, so if we add this to the description, it should be "Selected tab X".

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Tabstrip.cs]: /removed

y the "not announce" part drove me crazy - but ok it's not part of the original bug.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Tabstrip.cs [hunk 2]
```diff
@@ -345,36 +343,44 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
 		}
 
 		int focusedTab = -1;
+
+		int FocusedTab {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Tabstrip.cs]: This property is now redundant. Revert these changes

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Tabstrip.cs]: It doesn't make any sense to have a private property which is similar to an auto property. Either removed `focusedTab` or remove the property alltogether. It's just noise and error prone. It's really easy to set either focusedTab and FocusedTab, and if it's no longer an autoproperty, it can break.

**Comment by netonjm** [comment]
The PR seems include 2 atomic changes, the FocusedTab refactoring and the fix accessibility fix. 
It would be great to separate to in case of regression detect easy the problem.

And related the accessibility fix in Accessible.SetRole (AtkCocoa.Roles.AXRadioButton, active ? "active tab" : "tab"); maybe we could change only the message instead set the role?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Tabstrip.cs [hunk 3]
```diff
@@ -543,11 +555,12 @@ public Tab (Tabstrip parent, string label, TabPosition tabPosition)
 
 			if (AccessibilityElementProxy.Enabled) {
 				Accessible = AccessibilityElementProxy.ButtonElementProxy ();
-				Accessible.SetRole (AtkCocoa.Roles.AXRadioButton, "tab");
 				Accessible.Title = label ?? "";
+				Accessible.SetRole (AtkCocoa.Roles.AXRadioButton, "tab");
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Tabstrip.cs]: This change is redundant since UpdateAccessibility does it.

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Tabstrip.cs]: removed - but was old code no idea why it moved :/

**Comment by sevoku** [comment]
@monojenkins backport release-8.4

**Comment by sevoku** [comment]
@monojenkins backport release-8.4

**Comment by garuma** [comment]
See https://github.com/mono/monodevelop/pull/8925#pullrequestreview-304215669

**Comment by mkrueger** [comment]
Comment such stuff on VSTS, it's not reflected there.
